.github/workflows/ci.yml

```diff
@@ -563,6 +563,43 @@ jobs:
   ############################### Configurations ###############################
   # Jobs that build IREE in some non-default configuration
   ##############################################################################
+  python_release_packages:
+    needs: setup
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
+        with:
+          submodules: true
+      - name: Build runtime wheels (Linux)
+        shell: bash
+        env:
+          packages: "iree-runtime iree-runtime-instrumented"
+          output_dir: "${{ github.workspace }}/bindist"
+          # Note when upgrading: Build just one Python version synced to our
+          # minimum. Note that 3.7 is the last with this naming convention.
+          # Look at the script for the full list.
+          override_python_versions: cp37-cp37m
+        run: |
+          ./build_tools/python_deploy/build_linux_packages.sh
+      - name: Build compiler wheels (Linux)
```

Comment:
This seems like conceptually a separate job rather than a separate step. It doesn't need any of the things from the first step. Then again, a separate job means the overhead of acquiring a runner, checkout, etc. Since this isn't the critical path, I don't object to keeping them fused, but if you do, could you add a note that this is manual fusion efficiency and there's not actually a dependency link? Man if only we could have a compiler to automatically fuse jobs :-P

Reply:
I'll add a comment. It was an overhead tradeoff to put them together, not load bearing: saves a few GB of docker image download and all of the cloning. The runtime package also gives a lot of early signal which is why I put it first (and outside of startup overhead, just takes a couple of minutes).

Reply:
If only the unwieldyness of the compiler wasn't the reason to need to fuse and unfuse jobs...